drivers/gpio/gpio_mcux.c

```diff
@@ -293,7 +293,8 @@ static uint32_t get_port_pcr_irqc_value_from_flags(const struct device *dev, uin
 #endif /* !(defined(CONFIG_PINCTRL_NXP_IOCON)) */
 
 #if (defined(FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT) && \
-		FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT)
+		FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT) \
+		||  (defined(FSL_FEATURE_PORT_HAS_NO_INTERRUPT) && FSL_FEATURE_PORT_HAS_NO_INTERRUPT)
```

Comment:
since these long expressions are repeated so many times, it would be nice if the driver had a macro like

#if (defined(FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT) && \
		FSL_FEATURE_GPIO_HAS_INTERRUPT_CHANNEL_SELECT)
#define GPIO_MCUX_HAS_INTERRUPT_CHANNEL_SELECT
#endif

for these things